src/FMBot.Bot/Commands/Guild/UserGuildSettingCommands.cs

```diff
@@ -299,5 +299,71 @@ public async Task BlockedUsersAsync()
             await ReplyAsync("", false, this._embed.Build());
             this.Context.LogCommandUsed();
         }
+
+
+        [Command("whoknowswhitelist", RunMode = RunMode.Async)]
+        [Summary("Configures the WhoKnows charts to only show members with a specified role.\n\n" +
+                 "After changing your whitelist setting, you will most likely need to index the server again for it to take effect.\n\n" +
+                 "To remove the current whitelist setting, use this command without an option")]
+        [Examples("whoknowswhitelist", "whoknowswhitelist royals", "whoknowswhitelist 423946236102705154")]
+        [Alias("wkwhitelist", "wkwl")]
+        [GuildOnly]
+        public async Task SetWhoKnowsWhitelistRoleAsync([Remainder] string roleQuery = null)
+        {
+            var serverUser = (IGuildUser)this.Context.Message.Author;
+            if (!serverUser.GuildPermissions.BanMembers && !serverUser.GuildPermissions.Administrator &&
+                !await this._adminService.HasCommandAccessAsync(this.Context.User, UserType.Admin))
+            {
+                await ReplyAsync(
+                    "You are not authorized to use this command. Only users with the 'Ban Members' permission, server admins or FMBot admins can use this command.");
+                this.Context.LogCommandUsed(CommandResponse.NoPermission);
+                return;
+            }
+
+            if (string.IsNullOrEmpty(roleQuery) || roleQuery.ToLower() == "remove" || roleQuery.ToLower() == "delete")
+            {
+                await this._guildService.SetGuildWhoKnowsWhitelistRoleAsync(this.Context.Guild, null);
+                await this._guildService.StaleGuildLastIndexedAsync(this.Context.Guild);
```

Comment:
Do we really need to stale it here? Since if the server has no whitelist role the field on GuildUsers will be ignored anyway

Reply:
The WKWL checks in SQL queries still looks at the `GuildUser.WhoKnowsWhitelisted` column regardless of guild settings